src/framework/step/mod.rs

```diff
@@ -0,0 +1,132 @@
+use crate::framework::Error;
+#[cfg(feature = "enable-serde")]
+use serde::{
+    de::{Error as SerdeError, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use std::fmt::Formatter;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Step {
+    Example(ExampleCircuit),
+    ModulusConversion(ModulusConversionDetails),
+}
+
+impl Step {
+    #[must_use]
+    pub fn to_path(&self) -> String {
```

Comment:
I wonder why it is not enough to just implement `Serialize, Deserialize` on `Step` and force the same for enclosed structs/enums?
https://docs.rs/axum/0.5.15/axum/extract/struct.Path.html

Reply:
let's talk about that. the implementation of `Serialize`/`Deserialize` here was a lot of boilerplate, and i wanted to only have to do that once (and rely on string parsing after that).

we could do `serialize_with` and `deserialize_with` to get around that tho. but even that is more boilerplate than what i've got here